### DISCUSSION TITLE
In birt how to choose the font dynamically based upon text which comes from the database to generate the pdf report

### DISCUSSION THREAD

**Discussion by Akshay-64** [eclipse-birt/birt#1244] [Q&A]
In My database data contain different languages like Japanese, Chinese ,Bengali , Hindi and other foreign  languages. Is there any one font which supports all the languages or is there any alternative to choose the font based on text dynamically by the birt . My report was working fine in excel and html but it was not working in pdf format. Please help out from this issue.

**Comment by hvbtup** [comment]
You can try Microsoft's "Arial Unicode" ttf font. However, the glyphs need more height in comparison to Arial.

**Comment by Akshay-64** [comment]
Could you please provide the download link for that ttf font so that there will no confusion between the fonts we choose

**Comment by hvbtup** [comment]
Please do your homework yourself.

**Comment by Akshay-64** [comment]
It's not the homework actually it was the task I got but I am facing some difficult to do this because the text was dynamic that contains different languages If know how to do this please me

**Comment by speckyspooky** [comment]
Try this link of Arial Unicode MS, perhaps it is what you want: https://www.download-free-fonts.com/details/88978/arial-unicode-ms

**Comment by Akshay-64** [comment]
I tired with above link but above font was not support hindi and bengali languages . If any other fonts you know which supports all languages . Thanks for the suggestion you gave

**Comment by speckyspooky** [comment]
One suggestion from my side would be if you have all different languages on your db,
then I would try to collect the fonts of the different languages - not to try 1 for All.
And then change the font information dynamically at this moment if you need the different language.

The idea is to map the language:
DE = Arial Unicode
US = Arial Unicode
RO = Arial Unicode
FR = Arial Unicode
Hindi = Font-01(?)
Japan = Font-02(?)

And according to your requirement, you can change the font family from Unicode to Hindi.
The simplest way would be to use a BIRT-CSS and later you can fetch the CSS and change the font-family to your necessary familiy.
This way can be used at report level or e.g. table level other levels/elements should be also possible.

I change the font family for example on CSS-report level because my customer companies are German companies but use different fonts due to her according to CI.

The advantage would be with this idea you can use different fonts.

**Reply by Akshay-64** [reply]
Thanks for the suggestion 
 In Internet I found the plugin which contains NL Pack ,By using that we can able to get pdf report with maximum all languages(Except  in hindi,Bengali).
https://archive.eclipse.org/birt/downloads/lang.php
I inserted NL pack in local machine(Windows) it was working fine. But when I am deploying to Linux, empty text was displaying in PDF 
Can I insert the NL pack in birt_runtime.jar ??

By adding Arial Unicode font I am facing same issue(locally working but when I was deploying I am facing the issue-empty characters ). I did changes in fontsConfig.xml(adding path for the font) and also I added fontsConfig_linux(adding path for the font) and I removed fontsConfig_pdf file(because changes are reflecting when there is no fontsConfig_pdf)

Any suggestions for the above problems was appreciated
